test/e2e/federated-namespace.go

```diff
@@ -0,0 +1,118 @@
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/kubernetes/federation/client/clientset_generated/federation_release_1_4"
+	"k8s.io/kubernetes/federation/pkg/federation-controller/util"
+	"k8s.io/kubernetes/pkg/api"
+	api_v1 "k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	namespacePrefix = "e2e-namespace-test-"
+)
+
+// Create/delete ingress api objects
+var _ = framework.KubeDescribe("Federation namespace [Feature:Federation]", func() {
+	f := framework.NewDefaultFederatedFramework("federation-namespace")
+
+	clusterClientSet := make(map[string]*federation_release_1_4.Clientset)
+
+	Describe("Namespace objects", func() {
+
+		BeforeEach(func() {
+			framework.SkipUnlessFederated(f.Client)
+			clusters := buildClustersOrFail_14(f)
+			for _, cluster := range clusters {
+				if _, found := clusterClientSet[cluster.Name]; !found {
+					clientset, err := util.GetClientsetForCluster(cluster)
+					if err != nil {
+						framework.Failf("Failed to create client for %s: %v", cluster.Name, err)
+					}
+					clusterClientSet[cluster.Name] = clientset
+				}
+			}
+		})
+
+		AfterEach(func() {
+			framework.SkipUnlessFederated(f.Client)
+			deleteAllTestNamespaces(f.FederationClientset_1_4)
+			for _, clientset := range clusterClientSet {
+				deleteAllTestNamespaces(clientset)
+			}
+		})
+
+		It("should be created and deleted successfully", func() {
+			framework.SkipUnlessFederated(f.Client)
+
+			ns := api_v1.Namespace{
+				ObjectMeta: api_v1.ObjectMeta{
+					Name: fmt.Sprintf("%s%d", namespacePrefix, time.Now().UnixNano()),
+				},
+			}
+			By(fmt.Sprintf("Creating namespace %s", ns.Name))
+			_, err := f.FederationClientset_1_4.Core().Namespaces().Create(&ns)
+			framework.ExpectNoError(err, "Failed to create namespace %s", ns.Name)
+
+			// Check subclusters if the namespace was create there.
+			err = wait.Poll(5*time.Second, 2*time.Minute, func() (bool, error) {
+				for _, client := range clusterClientSet {
+					_, err := client.Core().Namespaces().Get(ns.Name)
+					if err != nil {
+						return false, err
+					}
+					// TODO: Uncomment once the namespace controller is fully enabled.
+					// if item == nil {
+					// 	return false, nil
+					// }
+				}
+				return true, nil
+			})
+			// This is a dummy check, the above snippent must be enabled.
+			framework.ExpectNoError(err, "Not all namespaces created")
+
+			err = f.FederationClientset_1_4.Core().Namespaces().Delete(ns.Name, &api.DeleteOptions{})
+			framework.ExpectNoError(err, "Failed to delete namespace %s", ns.Name)
+		})
+
+	})
+})
+
+func deleteAllTestNamespaces(clientset *federation_release_1_4.Clientset) {
+	list, err := clientset.Core().Namespaces().List(api.ListOptions{})
+	if err != nil {
+		framework.Logf("Failed to get all namespaes: %v", err)
```

Comment:
The let's change the function to return an error and return an error here.

Currently you are silently dropping errors, which is bad.